### PR TITLE
[86bzgwguf][docs] Replaced icon with ButtonLink in input examples

### PR DESCRIPTION
**Pull request by sheila-semrush**
## Motivation and Context

Replaced interactive icons with ButtonLink instances in Input playground and examples (also in InputPhone examples).

## How has this been tested?

Manually.

## Screenshots (if appropriate):

image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [x] Nice improve.

## Checklist:

- [x] My code follows the code style of this project.
- [x] I have updated the documentation accordingly or it's not required.
- [ ] Unit tests are not broken.
- [ ] I have added changelog note to corresponding `CHANGELOG.md` file with planned publish date.
- [ ] I have added new unit tests on added of fixed functionality.

### CODE REVIEW

**Review by j-mnizhek — APPROVED** [review]
Looks good! 😎 Just brought small fixes from vale

**Review by ilyabrower — APPROVED** [review]

**Comment by ilyabrower** [comment]
> @ilyabrower I need some help fixing the tests. The "Input > Returns focus on Clear Addon keyboard press" test fails because the Clear control is now `Hint` instead of the `Close` icon, and my attempt to fix the test failed 😅

I fixed tests, but I would like to mention, that in the inline input, in the buttonLink.Addon we have a margin-bottom=2px and it's looks not too good. Maybe we should change that thing?
Screenshot 2024-09-18 at 21 24 35

**Comment by sheila-semrush** [comment]
> > @ilyabrower I need some help fixing the tests. The "Input > Returns focus on Clear Addon keyboard press" test fails because the Clear control is now `Hint` instead of the `Close` icon, and my attempt to fix the test failed 😅
> 
> I fixed tests, but I would like to mention, that in the inline input, in the buttonLink.Addon we have a margin-bottom=2px and it's looks not too good. Maybe we should change that thing? 

Discussed in a call that we could add a rule to remove the margin when there's a ButtonLink or Link in the Input.Addon, and it doesn't have any text. Would you like to add this in this PR or separate?

**Comment by ilyabrower** [comment]
> > > @ilyabrower I need some help fixing the tests. The "Input > Returns focus on Clear Addon keyboard press" test fails because the Clear control is now `Hint` instead of the `Close` icon, and my attempt to fix the test failed 😅
> > 
> > 
> > I fixed tests, but I would like to mention, that in the inline input, in the buttonLink.Addon we have a margin-bottom=2px and it's looks not too good. Maybe we should change that thing?
> 
> Discussed in a call that we could add a rule to remove the margin when there's a ButtonLink or Link in the Input.Addon, and it doesn't have any text. Would you like to add this in this PR or separate?

I'll add it here

File: semcore/input/src/style/input.shadow.css [hunk 1]
```diff
@@ -78,6 +78,11 @@ SAddon {
   padding: 0 var(--intergalactic-spacing-2x, 8px);
   height: 100%;
   color: var(--intergalactic-icon-secondary-neutral, #a9abb6);
+
+  button [data-ui-name="ButtonLink.Addon"],
+  a [data-ui-name="Link.Addon"] {
+    margin-bottom: 0;
```
Comment (sheila-semrush) [inline semcore/input/src/style/input.shadow.css:84]: @ilyabrower it's not button/a, it's a span that has that margin-bottom

Comment (ilyabrower) [reply semcore/input/src/style/input.shadow.css:84]: yes, but this span is inside button

Comment (sheila-semrush) [reply semcore/input/src/style/input.shadow.css:84]: sorry, I don't know wtf happened, I just looked at the branch and apparently I haven't pulled your changes, I don't know what I was testing on Friday %)

Now I see that the bottom margin is gone, but unfortunately it now looks misaligned (too low) in Firefox. Is there a way to fix it?
image

Comment (ilyabrower) [reply semcore/input/src/style/input.shadow.css:84]: Recheck it please. I changed display type and checked in that example and on ButtonLink page example

**Review by ilyabrower — APPROVED** [review]

**Pull request merged**